pkg/parser/parser.y

```diff
@@ -8811,7 +8811,12 @@ TableName:
 	}
 |	Identifier '.' Identifier
 	{
-		$$ = &ast.TableName{Schema: model.NewCIStr($1), Name: model.NewCIStr($3)}
+		schema := $1
+		if schema == "" {
```

Comment:
Alternatively, we could think about trimming the schema:

```
# this PR
tidb> create table ` `.t1(a int, b int);
ERROR 1049 (42000): Unknown database ' '

# MySQL
mysql>  create table ` `.t1(a int, b int);
ERROR 1102 (42000): Incorrect database name ' '
```

However, it's not necessary to be consistent with all types of errors returned, MySQL gives different errors for ```rename table t1 to ` `.t2``` and ```renname table t1 to `  abc`.t2```.

LGTM

Reply:
@bb7133 Thank you for the advice.  I added a fix to handle the white space schema.

I found a logic to validate schema which contains white space.
https://github.com/yoshikipom/tidb/blob/b2debdab385e36c0f8953ed7244473bbed1f3286/pkg/planner/core/preprocess.go#L798

Valid: `' a'` (start with white space)
Invalida: `'a '` (end with whitespace), `' '` (whitespace only), `''` (empty)

```
# before this fix
tidb>  create table ` `.t1(a int, b int);
ERROR 1049 (42000): Unknown database ' '

# after this fix (same error as mysql)
tidb> create table `  `.t1(a int, b int);
ERROR 1102 (42000): Incorrect database name '  '
```